src/midi/midi_fluidsynth.cpp

```diff
@@ -0,0 +1,245 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2002-2011  The DOSBox Team
+ *  Copyright (C) 2020       Nikos Chantziaras <realnc@gmail.com>
+ *  Copyright (C) 2020       The dosbox-staging team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "midi_fluidsynth.h"
+
+#if C_FLUIDSYNTH
+
+#include <clocale>
+#include <string>
+
+#include "control.h"
+
+// Fluidsynth uses libinstpatch which as of v1.1.3 has the nasty behavior of
+// changing the locale under our nose. We remember and restore the current
+// locale whenever this could happen.
+//
+// https://github.com/realnc/dosbox-core/issues/7
+// https://github.com/swami/libinstpatch/issues/37
```

Comment:
Is `libinstpatch` something we need to worry about regarding our `LIBS` link line?

Reply:
No :) It's a FluidSynth dependency, not ours. dosbox-core needed to care about this, because they bundle *all* dependencies in the repo and build everything statically with some specific flags that trigger the bug. Supposedly was fixed upstream already.